module/applications/actor/group-sheet.mjs

```diff
@@ -0,0 +1,198 @@
+      classes: ["dnd5e", "sheet", "actor", "group"],
+      template: "systems/dnd5e/templates/actors/group-sheet.hbs",
+      tabs: [{navSelector: ".tabs", contentSelector: ".sheet-body", initial: "members"}],
+      width: 620,
+      height: 620
+    });
+  }
+
+  /* -------------------------------------------- */
+  /*  Context Preparation                         */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async getData(options={}) {
+    const context = super.getData(options);
+
+    // Membership
+    const {sections, stats} = this.#prepareMembers();
+    Object.assign(context, stats);
+    context.sections = sections;
+
+    // Movement
+    context.movement = this.#prepareMovementSpeed();
+
+    // Biography HTML
+    context.descriptionFull = await TextEditor.enrichHTML(this.actor.system.description.full, {
+      secrets: this.actor.isOwner,
+      rollData: context.rollData,
+      async: true,
+      relativeTo: this.actor
+    });
+
+    // Summary tag
+    context.summary = game.i18n.format("DND5E.GroupSummary", {
+      members: [
+        stats.nMembers ? `${stats.nMembers} ${game.i18n.localize("DND5E.GroupMembers")}` : "",
+        stats.nVehicles ? `${stats.nVehicles} ${game.i18n.localize("DND5E.GroupVehicles")}` : ""
+      ].filterJoin(` ${game.i18n.localize("and")} `)
+    })
+    return context;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare membership data for the sheet.
+   */
+  #prepareMembers() {
+    const stats = {
+      currentHP: 0,
+      maxHP: 0,
+      nMembers: 0,
+      nVehicles: 0
+    }
+    const sections = {
+      character: {label: "Player Characters", members: []},
+      npc: {label: "Non-Player Characters", members: []},
+      vehicle: {label: "Vehicles", members: []}
+    };
+    for ( const member of this.object.system.members ) {
+      const m = {
+        actor: member,
+        id: member.id,
+        name: member.name,
+        img: member.img,
+        hp: {}
+      };
+
+      // HP bar
+      const hp = member.system.attributes.hp;
+      m.hp.current = hp.value + (hp.temp || 0);
+      m.hp.max = hp.max + (hp.tempMax || 0);
+      m.hp.pct = ((m.hp.current / m.hp.max) * 100).toFixed(2);
+      m.hp.color = dnd5e.documents.Actor5e.getHPColor(m.hp.current, m.hp.max).css;
+      stats.currentHP += m.hp.current;
+      stats.maxHP += m.hp.max;
+
+      if ( member.type !== "vehicle" ) stats.nMembers++;
+      else stats.nVehicles++;
+      sections[member.type].members.push(m);
+    }
+    for ( const [k, section] of Object.entries(sections) ) {
+      if ( !section.members.length ) delete sections[k];
+    }
+    return {sections, stats};
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare movement speed data for rendering on the sheet.
+   * @returns {{secondary: string, primary: string}}
+   */
+  #prepareMovementSpeed() {
```

Comment:
I'd say we should probably use the DMG's calculation rules to auto calculate travel pace based on the speed set here:

> - In 1 minute, you can move a number of feet equal to your speed times 10.
> - In 1 hour, you can move a number of miles equal to your speed divided by 10.
> - For daily travel, multiply your hourly rate of travel by the number of hours traveled (typically 8 hours).
> - For a fast pace, increase the rate of travel by one-third.
> - For a slow pace, multiply the rate by two-thirds.

Then we can display the pace values, rather than the base speed, perhaps even with a toggle between Fast/Normal/Slow pace.

Reply:
What I take away from this suggestion is that I can remove the "units" field from the movement object, since the same speed value works for different distance units according to the DMG rules.